### ISSUE TITLE
(un)assignment menu not shown

### ISSUE THREAD

**Issue by CircleCode** [mozilla/testpilot-containers#423]
steps to reproduce:

1. open a new work tab
2. navigate to site.example.com
3. right click > always open in this container
4. open a new (non container) tab
5. navigate to site.example.com
6. right click → **no always open in this container**
7. go to another tab, and come back to the bogus tab → the menu is here

**Comment by jonathanKingston** [comment]
@CircleCode just to confirm are you seeing the following at step 6.
![selection_558](https://cloud.githubusercontent.com/assets/338988/24702193/a14056ac-19f5-11e7-8327-abd34e71d494.png)

The tick indicates for me when it is assigned, perhaps that isn't intuitive?

**Comment by CircleCode** [comment]
at step 6, just after the tab has been redirected to the new container, I don't have this item in the menu.
after going to another tab and then back to this tab, this item is present.

Note : while trying ti reproduce it, it seems it only appears right after assignment, and in the first tab opened after assignment.

**Comment by jonathanKingston** [comment]
Agreed this is a bug then, I saw the same yesterday and wasn't able to get a clean Steps To Reproduce.

Thanks!

**Comment by jonathanKingston** [comment]
@kjozwiak is seeing this also, not sure how often

**Comment by kjozwiak** [comment]
I can reproduce this every single time and I'm seeing it pretty often while using the feature. Whenever I open a website that's been assigned to a specific container, the "Always Open in This Container" will be missing from the context menu. As @CircleCode mentioned above, switching tabs and than going back will fix the problem.

Sometimes I'm seeing this as soon as I've installed the containers experiment and want to assign a website to a specific container, example: 
* [https://youtu.be/246jXN2O0KI](https://youtu.be/246jXN2O0KI)

**STR being used:** (pretty similar to the STR added by @CircleCode)

* install the latest version of m-c
* install the [test pilot](https://testpilot.firefox.com/) & the [containers experiment](https://testpilot.firefox.com/experiments/containers)
* open a personal container via the "+" new tab button and load facebook.com
* right click on the page and select "Always Open in This Container"
* close the personal tab and load facebook.com in a non-container tab
* select "Remember my decision for this site" and click on "Take me there"
* close the personal tab once again
* load facebook.com within a non-container tab and right click on facebook.com once it loads in the personal container

You'll notice that the "Always Open in This Container" context entry will be missing. If you switch to a different tab and return, the context item will appear.

**Affected Platforms:**

* macSO 10.12.4 x64 - Reproduced
* Windows 10 Pro x64 VM - Reproduced
* Ubuntu 16.04.2 LTS x64 VM - Reproduced

**Comment by jonathanKingston** [comment]
I think this is related to #520

**Comment by grahamperrin** [comment]
Not reproducible for me with `7.3.0` added to Firefox `86.0.1` on FreeBSD `14.0-CURRENT`: 

![image](https://user-images.githubusercontent.com/192271/111099112-b79d3f80-853c-11eb-8d53-81dedd22b8c9.png)

* there's the option to clear the check box.

**Comment by dannycolin** [comment]
I can't reproduce in the latest version and the context menu has changed since this bug has been reported. @CircleCode can you still reproduce?
